### PR TITLE
Prevent parsing IP strings twice in some cases

### PR DESCRIPTION
**Pull request by lpellegr**

### CODE REVIEW

File: src/main/java/net/ripe/ipresource/IpAddress.java [hunk 1]
```diff
@@ -40,9 +40,10 @@ public static IpAddress parse(String s) {
     }
 
     public static IpAddress parse(String s, boolean defaultMissingOctets) {
-        if (findFirstDotOrColon(s) == '.') {
+        final char findFirstDotOrColonResult = findFirstDotOrColon(s);
```
Comment (lolepezy) [inline src/main/java/net/ripe/ipresource/IpAddress.java:43]: s/findFirstDotOrColonResult/firstDotOrColonResult/g ?

Comment (lpellegr) [reply src/main/java/net/ripe/ipresource/IpAddress.java:43]: I am not sure to understand. Do you want to rename the new local variable I introduced to `findFirstDotOrColon`?

Comment (ties) [reply src/main/java/net/ripe/ipresource/IpAddress.java]: I think he meant to rename the variable`firstDotOrColon` - at least, that would be clearer for me.

Comment (lpellegr) [reply src/main/java/net/ripe/ipresource/IpAddress.java]: OK, thanks. The last commit I made fixes this.

Comment (lolepezy) [reply src/main/java/net/ripe/ipresource/IpAddress.java]: Yes, exactly that. Thanks!

**Pull request merged**